src/middlewared/middlewared/plugins/reporting/netdata_configure.py

```diff
@@ -24,17 +25,21 @@ def netdata_state_location(self):
 
     @private
     def post_dataset_mount_action(self):
-        if os.path.exists(get_netdata_state_path()):
+        netdata_state_path = get_netdata_state_path()
+        if os.path.exists(netdata_state_path):
```

Comment:
This is prone to races and is a bit confusing. Why do we check for the existence of this path and return but down below we do `os.makedirs(..exist_ok=True)`? I think we should just remove the existence check, right? @anodos325 thoughts?

Reply:
Okay then change os.makedirs logic to have exist_ok=False and catch the DirectoryExists exception

Reply:
Sounds like a bug. I will work on a fix shortly.